lib/packwerk/parsed_constant_definitions.rb

```diff
@@ -51,6 +51,8 @@ def collect_local_definitions_from_root(node, current_namespace_path = [])
       end
 
       Node.each_child(node) { |child| collect_local_definitions_from_root(child, current_namespace_path) }
+    rescue Node::TypeError
+      nil
```

Comment:
I'm a bit concerned about this part. It looks like in the future, this could lead to parsing bugs being silently ignored, and the behavior of packwerk regressing.

From your tests, it looks like the case that you want to fix is dynamic namespacing (`self::SOME_CONSTANT`). The fact that packwerk chokes on that seems like a bug to me. Can we find a solution that fixes this exact bug, instead of introducing a clause that ignores all unknown nodes?

Reply:
I have removed the offending lines you point to and added the ability for packwerk to parse the `self::` syntax